src/Rules/PhpDoc/PhpDocLineHelper.php

```diff
@@ -0,0 +1,28 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\PhpDoc;
+
+use PhpParser\Node;
+use PHPStan\PhpDocParser\Ast\PhpDoc\PhpDocTagNode;
+
+class PhpDocLineHelper
+{
+
+	/**
+	 * This method returns exact line of e.g. `@param` tag in PHPDoc so that it can be used for precise error reporting
+	 * - exact position is available only when bleedingEdge is enabled
+	 * - otherwise, it falls back to given node start line
+	 */
+	public static function detectTagLine(Node $node, PhpDocTagNode $phpDocTagNode): int
```

Comment:
Any https://phpstan.github.io/phpdoc-parser/PHPStan.PhpDocParser.Ast.Node.html should be accepted here, not just tag.